File: leetcode3423/Maximum_Difference_Between_Adjacent_Elements_in_a_Circular_Array.java
import java.util.Arrays;

class Solution {
    public int maxAdjacentDistance(int[] nums) {
        if (nums == null || nums.length < 2) {
            return 0;
        }
        int res = Math.abs(nums[0] - nums[nums.length - 1]);

        for (int i = 0; i < nums.length - 2; i++) {
            res = Math.max(res, Math.abs(nums[i] - nums[i + 1]));
        }
        return res;
    }
}

public class Maximum_Difference_Between_Adjacent_Elements_in_a_Circular_Array {

    public static void main(String[] args) {

        Solution solution = new Solution();

        // Example 1
        int[] nums1 = { 1, 2, 4 };
        int result1 = solution.maxAdjacentDistance(nums1);
        System.out.println("Input: " + Arrays.toString(nums1));
        System.out.println("Output: " + result1);
        System.out.println("Expected: 3");
        System.out.println();

        // Example 2
        int[] nums2 = { -5, -10, -5 };
        int result2 = solution.maxAdjacentDistance(nums2);
        System.out.println("Input: " + Arrays.toString(nums2));
        System.out.println("Output: " + result2);
        System.out.println("Expected: 5");
    }
}

File: leetcode3423/Maximum_Difference_Between_Adjacent_Elements_in_a_Circular_Array.py
class Solution:
    def maxAdjacentDistance(self, nums):
        """
        Function to calculate the maximum absolute difference between adjacent elements
        in a circular array.
        """
        if not nums or len(nums) < 2:
            return 0

        res = abs(nums[0] - nums[-1])
        for i in range(len(nums) - 1):
            res = max(res, abs(nums[i] - nums[i + 1]))
        return res


if __name__ == "__main__":
    solution = Solution()

    # Example 1
    nums1 = [1, 2, 4]
    result1 = solution.maxAdjacentDistance(nums1)
    print(f"Input: {nums1}")
    print(f"Output: {result1}")
    print("Expected: 3")
    print()

    # Example 2
    nums2 = [-5, -10, -5]
    result2 = solution.maxAdjacentDistance(nums2)
    print(f"Input: {nums2}")
    print(f"Output: {result2}")
    print("Expected: 5")